src/sys/unix.rs

```diff
@@ -560,6 +621,35 @@ impl Socket {
         Ok(n as usize)
     }
 
+    #[cfg(not(target_os = "redox"))]
+    pub fn send_vectored(&self, bufs: &[IoSlice<'_>], flags: c_int) -> io::Result<usize> {
+        let mut msg: libc::msghdr = unsafe { std::mem::zeroed() };
+        // Must give a mutable pointer, but kernel will only read.
+        msg.msg_iov = bufs.as_ptr() as *mut libc::iovec;
+        msg.msg_iovlen = bufs.len().min(max_iov()) as _;
+
+        let n = syscall!(sendmsg(self.fd, &mut msg as *mut libc::msghdr, flags))?;
+        Ok(n as usize)
+    }
+
+    #[cfg(not(target_os = "redox"))]
+    pub fn send_to_vectored(
+        &self,
+        bufs: &[IoSlice<'_>],
+        flags: c_int,
+        addr: &SockAddr,
+    ) -> io::Result<usize> {
+        let mut msg: libc::msghdr = unsafe { std::mem::zeroed() };
```

Comment:
Nit: same as above for the `libc::msghdr`.

Reply:
Done :)